### PR TITLE
added archive package, added go.sum file, fixed error handling

### PR DESCRIPTION
**Pull request by owilliams99**
Still looking for a way tar the file without the additional directories

### CODE REVIEW

File: archive/tarfile.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	V "github.com/gspivey/drayage/volume"
+)
+
+//WriteToTar tars any found volumes
+func WriteToTar() (err error) {
+	volumes, err := V.ListVolumes()
+	for _, vol := range volumes {
+		dir, err := os.Open(vol.Mountpoint)
+		if err != nil {
+			return fmt.Errorf("can't open %v (did you use sudo?)", vol.Name)
```
Comment (gspivey) [inline archive/tarfile.go]: On your return "chain" the errors like in the example below:

```go
if err := dec.Decode(&val); err != nil {
    if serr, ok := err.(*json.SyntaxError); ok {
        line, col := findLine(f, serr.Offset)
        return fmt.Errorf("%s:%d:%d: %v", f.Name(), line, col, err)
    }
    return err
}
```

Something like this should work:

```go 
return fmt.Errorf("can't open %v (did you use sudo?) : %v ", vol.Name, err)
```

File: archive/tarfile.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	V "github.com/gspivey/drayage/volume"
```
Comment (gspivey) [inline archive/tarfile.go]: the word `volume` does not seem to be used. Why alias it to V?

File: archive/tarfile.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+			return fmt.Errorf("can't read directory")
+		}
+
+		destfile := vol.Name + ".tgz"
+
+		//
+		tarfile, err := os.Create(destfile)
+		defer tarfile.Close()
+
+		var fileWriter io.WriteCloser = tarfile
```
Comment (gspivey) [inline archive/tarfile.go]: Nothing to address: Fascinating. You passed each type of iowriter is acting as a "filter".cool

File: archive/tarfile_test.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package archive
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestWriteToTar(t *testing.T) {
```
Comment (gspivey) [inline archive/tarfile_test.go]: Nothing to address: We have been using the test capability just to run our code from our package. We should start writing some actual unit tests to go with the code

**Review by gspivey — CHANGES_REQUESTED** [review]
Please address two of my comments. Overall this looks good. I expect `WriteToTar` to change soon. its going to need some parameters to change how it operates. For example how would you only tar a specific volume? You don't need to update that part now but something to think about.

**Comment by gspivey** [comment]
> Still looking for a way tar the file without the additional directories

Don't worry about this part for now. There are enough other things we can work on before worrying about this. Write an issue for this problem and we can address it later.

**Comment by owilliams99** [comment]
I have updated the WriteToTar function and have gotten the tar command working with drayage

**Review by gspivey — APPROVED** [review]

**Pull request merged**